164829216 functionality to tag articles

Sm0k3s opened the pull request:
#### Description
Users can tag their articles. This comes in handy when searching as one can get all articles that belong to a specific tag.

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [x] End to end
- [x] Integration

#### Checklist:
- [x] My code follows the style guide for this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### How should this be manually tested?
1.  Clone the repository 
```https://github.com/andela/ah-centauri-backend.git```
2. On completing of cloning
```bash 
cd ah-centauri-backend
git checkout feature/164829218-search-filter-on-articles
```
3. Rename the .env.example file into .env and update the parameters with your own keys.

> For Centauri team members you can use the keys shared through slack

4. Start the development server and send the following through postman

**Get articles by tags**
```
GET /api/articles/q?tags=news,tech,world

A user retrieves all articles that contain the tags specified
```


#### Pivotal Tracker
[#164829216](https://www.pivotaltracker.com/n/projects/2321088/stories/164829216)

Sm0k3s committed:
feature(tag-articles)-enable-users-to-tag-articles

- implements tagging feature in articles
- adds test for tagging feature
- rebased from develop

[Delivers #164829216]

koitoror reviewed (approved):
Looks good.

ValerieRono reviewed (approved):
looks good to me Steve

oagutu commented on authors/apps/articles/serializers.py:
```diff
@@ -30,32 +42,37 @@ def to_representation(self, instance):
-            instance.description)
+        instance.description = validated_data.get('description', instance.description)
+        instance.tags.set(*tags)
+
```
This replaces the tags list rather than update it, ie. if I add/update a single tag to an article the tags original list will be replaced by the new single element list. Is this intended?

ValerieRono replied:
if i may, with regards to whether one is able to update an article on his branch, it is possible,

Sm0k3s replied:
It wasn't intended.
I have implemented this feedback and confirmed that I can update the tags without removing the previous ones.

oagutu reviewed (commented):
@Sm0k3s Good job on this. Please look at my comment and let back to me. Also could you confirm that you are actually able to update an article on your branch

Sm0k3s committed:
feature(tag-articles)-enable-users-to-tag-articles

- implements feedback
- fix can update without removing previous tags

[Delivers #164829216]
